### PR TITLE
[AUTO] skip conformace test cases with dynamic rank for AUTO plugin

### PR DESCRIPTION
**Pull request by yangwang201911**
### Details:
 - enable test case skip for conformance test.
 - skip OVInferRequestDynamicTests test suit for AUTO as CPU and GPU didn't support dynamic rank now.

### Tickets:
 - CVS-130409

### CODE REVIEW

File: src/tests/functional/plugin/conformance/test_runner/conformance_infra/src/skip_tests_config.cpp [hunk 1]
```diff
@@ -18,7 +18,12 @@ const char *targetPluginName = "";
 const char *refCachePath = "";
 
 std::vector<std::string> IRFolderPaths = {};
-std::vector<std::string> disabledTests = {};
+std::vector<std::string> disabledTests = {
+
+    // CPU and GPU do not support dynamic rank
+    // Issue: 66778 and 98723
+    R"(.*OVInferRequestDynamicTests.*/*_targetDevice=(AUTO).*)",
+};
```
Comment (songbell) [inline src/tests/functional/plugin/conformance/test_runner/conformance_infra/src/skip_tests_config.cpp:26]: how about other dynamic cases not with dynamic rank? we should not disable all of them?

File: src/tests/functional/plugin/shared/include/behavior/ov_infer_request/infer_request_dynamic.hpp [hunk 2]
```diff
@@ -40,7 +40,8 @@ class OVInferRequestDynamicTests : public testing::WithParamInterface<OVInferReq
     void SetUp() override;
     bool checkOutput(const ov::Tensor& in, const ov::Tensor& actual);
 
-    std::shared_ptr<ov::Core> ie = utils::PluginCache::get().core();
+    // core initialization here will cause exception as meta plugins are not included in core.get_available_devices()
+    std::shared_ptr<ov::Core> ie;
```
Comment (songbell) [inline src/tests/functional/plugin/shared/include/behavior/ov_infer_request/infer_request_dynamic.hpp:44]: why?

Comment (yangwang201911) [reply src/tests/functional/plugin/shared/include/behavior/ov_infer_request/infer_request_dynamic.hpp:44]: https://github.com/openvinotoolkit/openvino/pull/23166

File: src/tests/functional/plugin/conformance/test_runner/conformance_infra/src/skip_tests_config.cpp (hunk 1 above)
Comment (yangwang201911) [reply src/tests/functional/plugin/conformance/test_runner/conformance_infra/src/skip_tests_config.cpp:25]: Update to skip the case of the dynamic rank in this test suit.

**Comment by peterchen-intel** [comment]
@iefode If this is the correct way for non-supporting test cases? These test cases can only be enabled for AUTO if CPU or GPU can support them.

File: src/tests/functional/plugin/conformance/test_runner/conformance_infra/src/skip_tests_config.cpp (hunk 1 above)
Comment (iefode) [reply src/tests/functional/plugin/conformance/test_runner/conformance_infra/src/skip_tests_config.cpp:25]: The conformance concept is not to use skips at all. We demonstrate real counters without hacks

**Review by iefode — CHANGES_REQUESTED** [review]
Can not be merged to master

**Pull request closed without merging**